nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/versioning/StandardVersioningService.java

```diff
@@ -299,19 +322,48 @@ public boolean isPreSaveDoingCheckOut(Document doc, boolean isDirty, VersioningO
     @Override
     public VersioningOption doPreSave(Document doc, boolean isDirty, VersioningOption option, String checkinComment,
             Map<String, Serializable> options) {
+        return doPreSave(null, doc, isDirty, option, checkinComment, options);
+    }
+
+    @Override
+    public VersioningOption doPreSave(CoreSession session, Document doc, boolean isDirty, VersioningOption option,
+            String checkinComment, Map<String, Serializable> options) {
         option = validateOption(doc, option);
         if (isPreSaveDoingCheckOut(doc, isDirty, option, options)) {
             doCheckOut(doc);
-            followTransitionByOption(doc, option);
+            followTransitionByOption(session, doc, options);
         }
         // transition follow shouldn't change what postSave options will be
         return option;
     }
 
     protected void followTransitionByOption(Document doc, VersioningOption option) {
+        followTransitionByOption(null, doc, Collections.emptyMap());
```

Comment:
I don't think the old method without `CoreSession` should be kept, let's not complexify the implementation. You should make `doPostSave` take a `CoreSession` too to be consistent, and remove the usage of the old methods without `CoreSession`.